pysparkle/pysparkle/data_quality/utils.py

```diff
@@ -143,3 +147,65 @@ def execute_validations(
     gx_validation_results = validator.validate()
 
     return gx_validation_results
+
+
+def publish_quality_results_table(spark: SparkSession, dq_path: str, datasource_name: str,
+                              results: list, data_quality_run_date: date) -> DataFrame:
+    """
+    Given a ValidatorResults object, this function
+    writes out the validation results for a given date
+    and datasource in a table and returns any failing validations
+
+    Args:
+        spark: Spark session.
+        dq_path: File path to write the results table to
+        datasource_name: Name of the datasource to be validated
+        results: List containing validator results from great expectations
+        data_quality_run_date: Date that the validations were executed
+    
+    Returns:
+        Dataframe containing failed validations
+    """
+    columns = StructType(
+        [StructField('DataQualityRunDate', DateType(), True),
```

Comment:
In silver we're using snake_case for columns, so probably should here too

Reply:
done